rfcs/experimental/blocked_nd_range_ctad/README.md

```diff
@@ -0,0 +1,219 @@
+```
+
+This deduction guide only participates in overload resolution if `N` is not equal to neither 2 nor 3 to disambiguate between `[1]` and `[2]`.
+See [separate section](#ambiguity-while-passing-the-single-braced-init-list-of-size-2-or-3) for more details.
+
+For service constructors `[3]` and `[4]`, the following guides are proposed:
+
+```cpp
+// [g4]
+template <typename Value, unsigned int N>
+blocked_nd_range(blocked_nd_range<Value, N>, split)
+-> blocked_nd_range<Value, N>;
+
+// [g5]
+template <typename Value, unsigned int N>
+blocked_nd_range(blocked_nd_range<Value, N>, proportional_split)
+-> blocked_nd_range<Value, N>;
+```
+
+From the specification perspective, such a deduction guides can be generated as implicit deduction guides, in the same manner as copy and move constructors.
+But the current oneTBB implementation, these deduction guides are not generated implicitly, so the explicit guides are required.
+Guides `[g4]` and `[g5]` are not proposed to be a part of the spec, only a part of oneTBB implementation.
+
+## Open Questions
+
+### Ambiguity while passing the single braced-init-list of size 2 or 3
+
+While using the CTAD with `blocked_nd_range`, there is an ambiguity between two approaches while using a single braced-init-list of size 2 or 3:
+
+```cpp
+blocked_nd_range range1({10, 20});
+blocked_nd_range range2({10, 20, 5});
+```
+
+Since the template arguments for `blocked_nd_range` are not specified, there can be two possible resolutions:
+1. Be interpreted as one-dimensional range _[10, 20)_ (with grainsize 1 or 5). In this case it should be deduced as `blocked_nd_range<int, 1>` and 
+   constructed using the constructor `[1]`.
+2. Be interpreted as two (or three) dimensional range _[0, 10)_, _[0, 20)_ (and _[0, 5)_). In this case it should be deduced as `blocked_nd_range<int, 3>` 
+   and constructed using the constructor `[2]`. 
+
+Since it is unclear which resolution should be chosen, current proposal is not to support such use-case in CTAD and require the user to either explicitly
+specialize the template arguments, or to use array or `blocked_range` type itself to initialize the object.
+
+### Passing single C-array object of size 2 or 3
+
+Another interesting issue that should be resolved, is passing the single C-array object of size 2 or 3 to the constructor:
+
+```cpp
+int array[2] = {100, 200};
+tbb::blocked_nd_range range(array);
+```
+
+Since the `blocked_range` is not constructible from C-array and the braced-init-list is not used, the user expects the range to be deduced as
+`blocked_nd_range<int, 2>` and the constructor `[2]` to be used.
+
+If we add one more explicit deduction guide to support the code above, the single braced-init-list of size 2 or 3 would also match on this guide.
+
+There are the following options how this issue can be resolved:
+* Add a new deduction guide to support the code above. The downside of this approach is that it makes the ambiguity, discussed in the
```

Comment:
My vision is that it would be better to pick that option since the explanation of the downside would not be surprise for the user and will be understandable from the deduction guides definition. I also think that construction of multi-dimensional ranges are mostly common use-case for blocked_nd_range.

The only thing that actually can be affected, is some generic code that can create both one and multi-dimensional ranges